src/idlistener.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * cookieRetriever.h
```

Comment:
Is there a name mistmatch?

Also the file name and function name are every generic. It would improve understandability if it whould somehow mention SAML somewhere.